### PR TITLE
Color settings

### PR DESCRIPTION
**Pull request by AmonShokhinAhmed**
Implemented color settings. Colors will get read from the settings.ini, which will automatically create the necessary menu elements.  To handle on the fly color changes (for when we allow the settings to be opened during gameplay) a global `ColorManager` was created. We now have to color objects using the functions `color_object_by_property` or `color_object_by_method` so they get registered properly. 

This means the workflow to create and use a new color is:
 - add the color to the `default-settings.ini` under the` [colors]` header
 - use them in code by calling `ColorManager.color_object_by_property(<color_name>, <object>, <property>)` or `ColorManager.color_object_by_methody(<color_name>, <object>, <method>, [<extra_args...>])`
 - everything else should happen automatically

Example: 
- `ColorManager.color_object_by_property("player_secondary_accent", _wall_particles.material_override, "albedo_color")`
- `ColorManager.color_object_by_method("player_main",_front.material_override, "set_shader_param", ["color"])`

### CODE REVIEW

**Comment by AmonShokhinAhmed** [comment]
Only undraft this after the playtest on the 16.12.2021 are finished. I don't want to cause any problems on main before that.

File: recursio-client/Level/CapturePoint.gd [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-onready var neutral_color = Color(Constants.get_value("colors", "neutral"))
+onready var neutral_color_name = "neutral"
 # TODO: better use main schema for all coloring?
-onready var player_color = Color(Constants.get_value("colors", "player_main"))
-onready var enemy_color = Color(Constants.get_value("colors", "enemy_main"))
-onready var player_in_capture_color = Color(Constants.get_value("colors", "player_ghost_primary_accent"))
-onready var player_captured_color = Color(Constants.get_value("colors", "player_ghost_main"))
-onready var enemy_in_capture_color = Color(Constants.get_value("colors", "enemy_ghost_primary_accent"))
-onready var enemy_captured_color = Color(Constants.get_value("colors", "enemy_ghost_main"))
+onready var player_color_name = "player_main"
```
Comment (kb173) [inline recursio-client/Level/CapturePoint.gd:6]: the `onready` could be removed from these since they're just strings now

Comment (kb173) [inline recursio-client/Level/CapturePoint.gd:6]: the `onready` could be removed from these since they're just strings now

**Review by kb173 — COMMENTED** [review]
Thanks! Looks great imo, I like the design of the ColorManager. I feel like the `name+"_"+accent_type+"_accent"` stuff appears often enough to potentially warrant separate functions (or an additional optional parameter to `color_object_by_method`, e.g. from an "Accent" enum) that encapsulate the String concatenations, but maybe that'd be overkill. (Not required for the merge, just a nice-to-have.) Just one very minor point other than that.

File: recursio-client/Level/CapturePoint.gd (hunk 1 above)
Comment (AmonShokhinAhmed) [reply recursio-client/Level/CapturePoint.gd]: Fixed

**Pull request merged**